modules/auxiliary/gather/query_ldap.rb

```diff
@@ -0,0 +1,188 @@
+          entries = perform_ldap_query(ldap, filter)
+
+        when 'ENUM_EXCHANGE_RECIPIENTS'
+          # Find Exchange Recipients with or without fax addresses.
+          filter = Net::LDAP::Filter.construct('(|(mailNickname=*)(proxyAddresses=FAX:*))')
+          entries = perform_ldap_query(ldap, filter)
+
+        when 'ENUM_GROUPS'
+          # Standard LDAP groups query, followed by trying to find AD security groups, then trying to find Linux groups.
+          # Filters combined to remove duplicates.
+          filter = Net::LDAP::Filter.construct('(|(objectClass=group)(objectClass=groupOfNames)(groupType:1.2.840.113556.1.4.803:=2147483648)(objectClass=posixGroup))')
+          entries = perform_ldap_query(ldap, filter)
+
+        when 'ENUM_ORGUNITS'
+          filter = Net::LDAP::Filter.construct('(objectClass=organizationalUnit)') # Find OUs aka Organizational Units
+          entries = perform_ldap_query(ldap, filter)
+
+        when 'ENUM_ORGROLES'
+          filter = Net::LDAP::Filter.construct('(objectClass=organizationalRole)') # Find OUs aka Organizational Units
+          entries = perform_ldap_query(ldap, filter)
+
+        when 'ENUM_PEOPLE'
+          filter = Net::LDAP::Filter.construct('(objectClass=organizationalPerson)') # Find people within an organization by Person entries.
+          entries = perform_ldap_query(ldap, filter)
+
+        when 'ENUM_USERS'
+          # Common LDAP user query, followed by a query for AD User records by account type.
+          # Finally, query for Linux accounts by objectClass.
+          #
+          # Doing this all in one query also prevents duplicate entires across multiple queries.
+          filter = Net::LDAP::Filter.construct('(|(objectClass=inetOrgPerson)(objectClass=user)(sAMAccountType=805306368)(objectClass=posixAccount)(objectClass=GsAccount)(objectClass=GsSIPUser))')
+          entries = perform_ldap_query(ldap, filter)
+        end
+      end
+    rescue Rex::ConnectionTimeout => e
+      print_error("Could not query #{datastore['RHOST']}! Error was: #{e.message}")
+      return
+    end
+
+    return if entries[1].nil?
+
+    columns = []
+    entries[1].each do |entry|
+      entry.attribute_names.each do |attribute|
+        columns << attribute.to_s
+      end
+    end
+    columns.uniq!
+    tbl = Rex::Text::Table.new(
+      'Header' => "#{action.name} Dump of #{peer}",
+      'Indent' => 1,
+      'Columns' => columns
+    )
+    entries[1].each do |entry|
+      data = []
+      columns.each do |col|
+        col = col.to_sym
+        if entry[col].nil? || entry[col].empty? || entry[col][0].empty?
+          data << nil
+        else
+          data << entry[col].join(' || ')
+        end
+      end
+      tbl << data
+    end
+    print_status(tbl.to_s)
+  rescue Net::LDAP::Error => e
+    print_error("#{e.class}: #{e.message}")
+  end
```

Comment:
Should consolidate this with `Rex::ConnectionTimeout` by either placing both of them here, or moving them both to the begin ensure block you have since neither have anything to do with the table generation.

Reply:
Decided to consolidate both earlier in the code where the networking functionality was and remove this line since by the time you hit this part of the code you are in the table generation as you correctly noted, and catching these sorts of errors doesn't make as much sense.

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/16598/commits/f833a5cb15b9ba4fe1b29a30cf542a2c5a30325b